src/DotLiquid/Template.cs

```diff
@@ -27,32 +27,52 @@ public class Template
 	{
 		public static INamingConvention NamingConvention;
 		public static IFileSystem FileSystem { get; set; }
-		private static Dictionary<string, Type> Tags { get; set; }
+        private static Dictionary<string, Tuple<ITagFactory, Type>> Tags { get; set; }
         private static readonly Dictionary<Type, Func<object, object>> SafeTypeTransformers;
 		private static readonly Dictionary<Type, Func<object, object>> ValueTypeTransformers;
 
 		static Template()
 		{
 			NamingConvention = new RubyNamingConvention();
 			FileSystem = new BlankFileSystem();
-			Tags = new Dictionary<string, Type>();
+			Tags = new Dictionary<string, Tuple<ITagFactory, Type>>();
             SafeTypeTransformers = new Dictionary<Type, Func<object, object>>();
 			ValueTypeTransformers = new Dictionary<Type, Func<object, object>>();
 		}
 
 		public static void RegisterTag<T>(string name)
 			where T : Tag, new()
 		{
-			Tags[name] = typeof(T);
+            var tagType = typeof(T);
```

Comment:
Thanks @laedit - I've added you to the dotliquid organisation. Any time you can put into triaging pull requests and issues would be gratefully received!